doc/specs/reward_protocol.md

```diff
@@ -0,0 +1,202 @@
+
+```mermaid
+sequenceDiagram
+
+ProofTx->>+WorldState: IsClaimSubmissionWindowEnded()
+
+WorldState->>+PseudorandomSelection:Select(Blockheader,SessionHeader,TotalProofs)
+
+PseudorandomSelection-->>-WorldState: SelectedIndex
+
+WorldState-->>-ProofTx: SelectedIndex
+
+ProofTx->>+MSITree:GenerateMerkleProof(SelectedIndex)
+
+MSITree->>+RelayProofs:GetLeaf(SelectedIndex)
+
+RelayProofs-->>-MSITree:Leaf
+
+MSITree-->>-ProofTx: Leaf & MerkleProof
+
+ProofTx->>+WorldState: SubmitProofTx()
+
+WorldState->>WorldState: ValidateProofTx()
+
+WorldState-->>-ProofTx: IfValid() -> Mint POKT
+```
+
+### Pseudorandom Selection Of Proof Index
+Similar to the Session Protocol, the Pseudorandom Selection algorithm uses the BlockHash of `SessionEndHeight`+`ClaimSubmissionWindow` block as seed data for the Pseudorandom Selection.
+
+```go
+headerHash := sessionHeader.HashString()  
+pseudoGenerator := pseudorandomGenerator{blockHash, headerHash}
+r, err := json.Marshal(pseudoGenerator)  
+...
+PseudorandomSelection(totalRelays, Hash(r))
+```
+
+To increase security and to tailor unique index selections for each Node, the seed data is also calculated from the SessionHeader.
+
+```go
+func PseudorandomSelection(max sdk.BigInt, hash []byte) (index sdk.BigInt) {  
+   // first 8 bytes of the seedHash
+  intHash := sdk.NewIntFromBigInt(new(big.Int).SetBytes(hash[:8]))  
+   // modulo the selection against the maximum of the range
+  return intHash.Mod(max)  
+}
+```
+
+*NOTE: The algorithm must be **pseudo**random because it must also be deterministic for the network to have Consensus on the agreed upon index.*
+
+### Proof of the Claim
+
+Once the Pseudorandom index is revealed to the network, the final step is for the Service Node to submit a `Proof Transaction` to the network to satisfy:
+> the protocol is able to use a single randomly selected leaf from the tree and use that as proof of the entire set **as long as there exists a MSI Branch from the selected leaf to the given Merkle Root**
+
+Using the previously generated MSI Tree, the Service Node is able to generate a MSI Branch from the Pseudorandomly Selected Leaf to the Merkle Root.
+
+Including the Leaf and the MSI Branch, the Service Node is able to complete the lifecycle by submitting a **Proof Transaction** to the network.
+
+```go
+type MsgProof interface {  
+   MerkleBranch()  // the MSI Branch
+   Leaf()          // the pseudrandomly selected leaf
+   EvidenceType()  // relay (or challenge)
+}
+```
+
+
+### Reward
+
+Like any Transaction, the proof is validated by all who process blocks.
+
+In order to determine if a ProofTx is valid, several Validations must occur.
+
+```go
+func ValidateProofTx(proof MsgProof) error {  
+  // find corresponding claim
+  ...
+  // validate session exists
+  ...
+  // generate pseudorandom index and validate against leaf
+  ...
+  // validate the merkle proof  
+  ...
+  // validate the leaf
```

Comment:
What does this entail? I'm presuming this means comparing values, but can you specify which ones?